File: 자릿수더하기/javascript/20221018/solution.js
// eslint-disable-next-line import/prefer-default-export
export function split(n) {
  const number = n.toString().split('');

  return number;
}

export function sum(arr) {
  return arr.map((i) => parseInt(i, 10)).reduce((acc, x) => acc + x);
}

export function solution(n) {
  return sum(split(n));
}

File: 자릿수더하기/java/20221018/src/Solution.java
import java.util.Arrays;
import java.util.stream.Stream;

public class Solution {
    public int solution(int n) {
        int answer = sum(n);

        return answer;
    }

    public int sum(int i) {
        return Stream.of(Integer.toString(i).split(""))
                .mapToInt(number -> Integer.parseInt(number))
                .sum();
    }
}
